From: Peter Eisenmann <p3732@getgoogleoff.me>
Date: Fri, 15 Nov 2024 21:50:19 -0400
Subject: listbase: Factor out item select action activation

This is in preparation of reusing this function in the next comit.

Origin: https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/6354
Bug: https://gitlab.gnome.org/GNOME/nautilus/-/issues/2618
Bug: https://gitlab.gnome.org/GNOME/nautilus/-/issues/3078
Bug-Ubuntu: https://launchpad.net/bugs/2066062
---
 gtk/gtklistbase.c | 25 ++++++++++++++++++-------
 1 file changed, 18 insertions(+), 7 deletions(-)

diff --git a/gtk/gtklistbase.c b/gtk/gtklistbase.c
index 68682e8..ea3a57c 100644
--- a/gtk/gtklistbase.c
+++ b/gtk/gtklistbase.c
@@ -448,6 +448,23 @@ gtk_list_base_select_item (GtkListBase *self,
                                       0, 0);
 }
 
+static void
+activate_listitem_select_action (GtkListBasePrivate *priv,
+                                 guint               pos,
+                                 gboolean            modify,
+                                 gboolean            extend)
+{
+  GtkListTile *tile;
+
+  tile = gtk_list_item_manager_get_nth (priv->item_manager, pos, NULL);
+
+  /* We do this convoluted calling into the widget because that way
+   * GtkListItem::selectable gets respected, which is what one would expect.
+   */
+  g_assert (tile->widget);
+  gtk_widget_activate_action (tile->widget, "listitem.select", "(bb)", modify, extend);
+}
+
 /*
  * gtk_list_base_grab_focus_on_item:
  * @self: a `GtkListBase`
@@ -506,13 +523,7 @@ gtk_list_base_grab_focus_on_item (GtkListBase *self,
 
   if (select)
     {
-      tile = gtk_list_item_manager_get_nth (priv->item_manager, pos, NULL);
-
-      /* We do this convoluted calling into the widget because that way
-       * GtkListItem::selectable gets respected, which is what one would expect.
-       */
-      g_assert (tile->widget);
-      gtk_widget_activate_action (tile->widget, "listitem.select", "(bb)", modify, extend);
+      activate_listitem_select_action (priv, pos, modify, extend);
     }
 
   return TRUE;
